lib/puppet/type/postgresql_conf.rb

```diff
@@ -2,40 +2,36 @@
 
 Puppet::Type.newtype(:postgresql_conf) do
   @doc = 'This type allows puppet to manage postgresql.conf parameters.'
+
   ensurable
 
-  newparam(:name) do
-    desc 'A unique title for the resource.'
-    newvalues(%r{^[\w.]+$})
+  def self.title_patterns
+  [
+    [ /^(.+\.conf):([\w.]+)$/m, [ [:target], [:key] ] ],
+    # TODO: make target optional
+    #[ /^([\w.]+)$/m, [ [:key] ] ],
+  ]
   end
 
-  newparam(:key) do
-    desc 'The Postgresql parameter to manage.'
+
+  newparam(:key, namevar: true) do
```

Comment:
I kept `key` instead of `name` to force me to implement `ParsedFile` agnostic of the `name` parameter. At this point I think it could revert back to `name`.